### ISSUE TITLE
potential security issue: ecall_SSL_read using [user_check] 

### ISSUE THREAD

**Issue by wyann22** [lsds/TaLoS#20]
@Similar to issue13, i think it is necessary to add a bound-checking function in ecall_SSL_read. 
Image a TLS client which can fetch website contents of any server, we can transfer a pointer pointing to enclave memory like this:  
`bytes = SSL_read(ssl, (char*)(secret_addr), sizeof(buf)); `
So the secret memory will be overwritten to the website content like below:
![image](https://user-images.githubusercontent.com/44711240/63899542-1d76b480-c9f5-11e9-87a1-3cf08339bede.png)
Moreover, if the host is not google but controlled by attackers, the secret memory value can be overwritten to arbitrary value like below. (the left terminal is a server controlled by attacker)
![image](https://user-images.githubusercontent.com/44711240/63899785-1ac88f00-c9f6-11e9-8b2e-c9c50891089b.png)
Basically, attackers can modify to any memory in enclave to any value. 
Thanks for guidance from @david-oswald 
#18

**Comment by david-oswald** [comment]
I think this issue should be merged with #18

**Comment by plaublin** [comment]
Closed because duplicate of #18
